lib/topological_inventory/providers/common/operations/source.rb

```diff
@@ -0,0 +1,140 @@
+require "topological_inventory/providers/common/logging"
+require "active_support/core_ext/numeric/time"
+require "topological_inventory/providers/common/operations/sources_api_client"
+
+module TopologicalInventory
+  module Providers
+  module Common
+    module Operations
+      class Source
+        include Logging
+
+        STATUS_AVAILABLE, STATUS_UNAVAILABLE = %w[available unavailable].freeze
+
+        ERROR_MESSAGES = {
+          :authentication_not_found => "Authentication not found in Sources API",
+          :endpoint_not_found       => "Endpoint not found in Sources API",
+        }.freeze
+
+        LAST_CHECKED_AT_THRESHOLD = 5 # minutes
+
+        attr_accessor :params, :request_context, :source_id
+
+        def initialize(params = {}, request_context = nil)
+          self.params          = params
+          self.request_context = request_context
+          self.source_id       = params['source_id']
+        end
+
+        def availability_check
+          return if params_missing?
+
+          return if checked_recently?
+
+          status, error_message = connection_status
+
+          update_source_and_endpoint(status, error_message)
+
+          logger.availability_check("Completed: Source #{source_id} is #{status}")
+        end
+
+        private
+
+        def required_params
```

Comment:
Should this be a constant?

Reply:
I don't think that overwritable method with list of required params is the best candidate for constant